driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java

```diff
@@ -221,8 +221,14 @@ private List<BsonElement> getExtraElements(final SessionContext sessionContext)
         if (sessionContext.getClusterTime() != null) {
             extraElements.add(new BsonElement("$clusterTime", sessionContext.getClusterTime()));
         }
-        if (sessionContext.hasSession() && responseExpected) {
-            extraElements.add(new BsonElement("lsid", sessionContext.getSessionId()));
+        if (sessionContext.hasSession()) {
```

Comment:
This is really the crux of the whole change.  A few notes:

* All application-initiated messages now have a session, so the only time this is false is for SDAM-related connection
* This seems a bit late to be checking for unsupported explicit sessions, but it's the only place to put it that wouldn't result in code duplication, since sync and async paths all end up here
* In practice, almost every server that the driver could actually connect to (3.6+) supports sessions, so the only time `isSessionSupported` will return false is a 3.6 server in 3.4 FCV mode, or a mongocryptd server.